src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -93,6 +96,50 @@
   (instance? X509CRL x))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ValidX500Name
+  (schema/pred valid-x500-name?))
+
+(def Extension
+  "A map containing all the fields required to define an extension."
+  {:oid schema/Str
+   :critical (schema/pred #(not (nil? %)))
+   :value (schema/pred #(not (nil? %)))})
+
+(def ExtensionList
+  [Extension])
+
+(def CertOrCSR
+  (schema/cond-pre X509Certificate PKCS10CertificationRequest))
```

Comment:
I just missed updating the other `either`. `cond-pre` is the recommended replacement for `either`, which is deprecated.